crypto/provider_conf.c

```diff
@@ -244,7 +244,10 @@ static int provider_conf_load(OSSL_LIB_CTX *libctx, const char *name,
         else if (strcmp(confname, "module") == 0)
             path = confvalue;
         else if (strcmp(confname, "activate") == 0)
-            activate = 1;
+            if ((strcmp(confvalue, "1") == 0)
+                || (strcmp(confvalue, "yes") == 0)
+                || (strcmp(confvalue, "true") == 0))
+                activate = 1;
```

Comment:
What about an empty value? Or `activate = ignored value`, or what about uppercase or mixed case? I think it is plausible that someone when reading the existing documentation could set it to an empty value or something bogus. I agree that `0`, `no`, `false` can be regarded as false value but...

At least this deserves a CHANGES.md entry and it is of course eligible only for 3.3.

Reply:
Why not allow `activate` without assignment as synonym for `activate=true`?

Reply:
> Why not allow `activate` without assignment as synonym for `activate=true`?

That would be a config syntax change, which I believe is much bigger than the scope of this PR.

Reply:
I agree that this is only forward-release elligible, and I'm happy to add a CHANGES.md entry.

The documentation that I added expressly indicates that ony  the values 1, true and yes will be accepted to activate the provider, and that all other values, including changes in case, and  the omission of a value will not activate the provider.  I agree that that might confuse some people, but I think this will confuse them less than what we have currently.

Reply:
yeah, we could add activate = on, but, if the goal is to be explicit in what activates a provider, we need to settle on a complete set and let that be that. The rule I've been roughly following is: does the converse of the term clearly imply disabling, i.e. we detect activate = yes as an activation because activate = no clearly implies linguistic lack of activation, 1/0 do the same, as do true/false.  on/off seems to be applicable, but an empty token does not because a non-empy token does not imply a lack of activation